testsuite/mpitests/test_connect_array_fixed_indegree_mpi.sli

```diff
@@ -0,0 +1,117 @@
+/*
+ *  test_connect_array_fixed_indegree_mpi.sli
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+
+/* BeginDocumentation
+   Name: testsuite::test_connect_array_fixed_indegree_mpi - SLI script to test connection with parameter arrays and connection rule fixed_indegree, with hybrid MPI and thread parallelism
+
+   Synopsis: (test_connect_array_fixed_indegree_mpi) run
+
+   Description:
+
+   Tests with hybrid MPI and thread parallelism the possibility to use
+   parameter arrays to define connection weights and/or delays in conjunction
+   with the connection rules fixed_indegree
+
+   FirstVersion: August 2016
+   Author: Bruno Golosio
+   SeeAlso: Connect
+ */
+
+(unittest) run
+/unittest using
+
+is_threaded not { exit_test_gracefully } if
+
+[2 3] % check for 2, 3 processes
+{
+
+/N 20 def          % number of neurons in each subnet
+/K 5 def           % number of connections per neuron
+/VP_per_MPI 2 def  % number of virtual processes per MPI process
+
+ResetKernel
+NumProcesses /num_mpi_procs Set  % gets number of MPI processes
+
+0 <<
+     /total_num_virtual_procs num_mpi_procs VP_per_MPI mul
```

Comment:
You can also set `/local_num_threads 2` directly, then NEST will do the math for you. That is the recommended approach.